Dockerfile

```diff
@@ -27,30 +31,58 @@ COPY ./src/pkgagent/mod_deps ./src/pkgagent/
 COPY ./src/scheduler/mod_deps ./src/scheduler/
 COPY ./src/ununpack/mod_deps ./src/ununpack/
 COPY ./src/wget_agent/mod_deps ./src/wget_agent/
-COPY ./install/scripts/php-conf-fix.sh ./install/scripts/php-conf-fix.sh
-COPY ./utils/install_composer.sh ./utils/install_composer.sh
+
+RUN mkdir -p /fossology/dependencies-for-runtime \
+ && cp -R /fossology/src /fossology/utils /fossology/dependencies-for-runtime/
 
 RUN DEBIAN_FRONTEND=noninteractive apt-get update \
- && DEBIAN_FRONTEND=noninteractive /fossology/utils/fo-installdeps --everything -y \
- && rm -rf /var/lib/apt/lists/* \
- && /fossology/install/scripts/php-conf-fix.sh --overwrite \
- && /fossology/utils/install_composer.sh
+ && DEBIAN_FRONTEND=noninteractive /fossology/utils/fo-installdeps --build -y \
+ && rm -rf /var/lib/apt/lists/*
 
 COPY . .
 
-RUN make install && make clean
+RUN /fossology/utils/install_composer.sh
 
-# the database is filled in the entrypoint
-RUN /usr/local/lib/fossology/fo-postinstall --agent --common --scheduler-only --web-only --no-running-database
+RUN make && make install && make clean
```

Comment:
One suggestion, this statement can be reduced to
`RUN make install clean`

Reply:
The split in three commands ensures that the installation runs only after a successful compilation and so on. This is necessary since we recursive call multiple Makefiles.

Reply:
The `install` recipe already have the dependency on `all` recipe (which is called when simply calling `make`). So `install` will only run if compilation (i.e. `all`) is completed successfully. And then you can chain the `clean` recipe in the same command.

Reply:
Implemented.